lua/telescope/previewers.lua

```diff
@@ -98,6 +134,47 @@ local cat_maker = function(filename, _, start, _)
   end
 end
 
+local get_buffer = function(opts, status)
+  local bufnr = -1
+  if opts.path then
+    local ft = determine_filetype(opts.path)
+    bufnr = vim.fn.bufadd('')
```

Comment:
I would use `vim.api.nvim_create_buf` and make it a scratch and unlisted buffer.

Make sure that we wipe it when it's hidden.